native/dgetf2.go

```diff
@@ -0,0 +1,63 @@
+package native
+
+import (
+	"math"
+
+	"github.com/gonum/blas/blas64"
+)
+
+// Dgetf2 computes the LU decomposition of the m×n matrix a.
+// The LU decomposition is a factorization of a into
+//  A = P * L * U
+// where P is a permutation matrix, L is a unit lower triangular matrix, and
+// U is a (usually) non-unit upper triangular matrix. On exit, L and U are stored
+// in place into a.
+//
+// ipiv is a permutation vector. It indicates that row i of the matrix was
+// changed with ipiv[i]. ipiv must have length at least min(m,n), and will panic
+// otherwise.
+//
+// Dgetf2 returns whether the matrix a is singular. The LU decomposition will
+// be computed regardless of the singularity of A, but division by zero
+// will occur if the false is returned and the result is used to solve a
+// system of equations.
+func (Implementation) Dgetf2(m, n int, a []float64, lda int, ipiv []int) bool {
+	mn := min(m, n)
+	checkMatrix(m, n, a, lda)
+	if len(ipiv) < mn {
+		panic(badIpiv)
+	}
+	if m == 0 || n == 0 {
+		return true
+	}
+	bi := blas64.Implementation()
+	sfmin := dlamchS
+	ok := true
+	for j := 0; j < mn; j++ {
+		// Find a pivot and test for singularity.
+		jp := j + bi.Idamax(m-j, a[j*lda+j:], lda)
+		ipiv[j] = jp
+		if a[jp*lda+j] == 0 {
+			ok = false
+		} else {
+			// Swap the rows if necessary.
+			if jp != j {
+				bi.Dswap(n, a[j*lda:], 1, a[jp*lda:], 1)
+			}
+			if j < m-1 {
+				aj := a[j*lda+j]
+				if math.Abs(aj) >= sfmin {
+					bi.Dscal(m-j-1, 1/aj, a[(j+1)*lda+j:], lda)
+				} else {
+					for i := 0; i < m-j+1; i++ {
```

Comment:
The loop in fortran is `1,m-j` which is `1 <= i <= m-j`; shifted to 0-based `0 <= i <= m-j-1` which is `0 <= i < m-j`. So no adjustment at all. Am I missing something?